addon/services/activity-watcher.ts

```diff
@@ -0,0 +1,168 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+
+import { Observable } from '@upfluence/hyperevents/helpers/observable';
+import EventsService, { ResourceEvent, prefixPath } from '@upfluence/hyperevents/services/events-service';
+import ToastService, { ToastOptions } from '@upfluence/oss-components/services/toast';
+
+type NotificationEvent = {
+  resource: string;
+  payload: {
+    notification_type: string;
+    timestamp: number;
+    data: any;
+  };
+};
+
+type RenderedNotification = {
+  title: string;
+  message: string;
+  type: string;
+};
+
+function renderNotificationMessage(message: string): RenderedNotification {
+  return {
+    title: `<i class="toast-title__icon upf-icon upf-icon--messages"></i>`,
+    message: message,
+    type: 'info'
+  };
+}
+
+function renderNotificationErrorMessage(message: string): RenderedNotification {
+  return {
+    title: `<i class="fa fa-info-circle" aria-hidden="true"></i>`,
+    message: message,
+    type: 'error'
+  };
+}
+
+function renderNotificationMessageWithAvatar(avatarUrl: string, message: string): RenderedNotification {
+  return {
+    title: `<img class="toast-title__avatar" src="${avatarUrl}" />
+     <i class="toast-title__icon upf-icon upf-icon--messages"></i>`,
+    message: message,
+    type: 'info'
+  };
+}
+
+type NotificationRendererMap = {
+  [key: string]: (data: any) => RenderedNotification
+}
+
+const renderersByNotificationType: NotificationRendererMap = {
```

Comment:
Not exactly what I meant when I was talking about a dictionnary (I was suggesting something less verbose ; this is more or less the same code but in a const outside of the service. I'm still down to have a look with you.

Reply:
TELL ME WHERE AND WHEN, AND I'LL WE BE HERE